### PR TITLE
DSND-2232: Created a base project for filing-history-data-api

### PR DESCRIPTION
**Pull request by johncook-ch**
## Describe the changes
* Java 21 and Spring Boot 3.2.0
* Includes structured logging, basic unit and integration tests and Spring Actuator to implement a basic health check
* Updated README to remove Kafka-related settings
* Updated the PR template

### Related Jira tickets
[DSND-2232](https://companieshouse.atlassian.net/browse/DSND-2232)

### General
- [X] Is the PR as small as it can be?
- [X] Has the Jira ticket been updated with any relevant comments?
- [X] Is the `README` up to date?
- [X] Has the `POM` been updated for the latest versions of dependencies?
- [X] Has the code been double-checked against `main`?
- [X] Does the code adhere standards in this repository, including SonarQube checks?

### Testing
- [X] Do the code changes have unit and integration tests with code coverage > 80%?
- [X] Has the code been tested locally and/or passed the API Karate tests on Tilt?
- [n/a] Have mandatory manual test cases been run?
- [n/a] Are extra Karate tests required?
- [n/a] Are all the test data resources up to date?

### Release preparation
_Where possible, add links to the respective Jira tickets when an item is checked_
- [n/a] Have these changes been included in a release ticket?
- [n/a] Are changes required to the `docker-chs-development` deployment or test data?
- [n/a] Are any changes required to the environment added to `chs-configs`?

## Notes to the tester
None

[DSND-2232]: https://companieshouse.atlassian.net/browse/DSND-2232?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/filinghistory/api/interceptor/RequestLoggingInterceptor.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        DataMapHolder.initialise(Optional
+                .ofNullable(request.getHeader(REQUEST_ID.value()))
+                .orElse(UUID.randomUUID().toString()));
+        return true;
+    }
+
+    @Override
+    public void postHandle(@Nonnull HttpServletRequest request,
+            @Nonnull HttpServletResponse response,
+            @Nullable Object handler, ModelAndView modelAndView) {
```
[the remark this answers is not in the document]
Comment (johncook-ch) [reply src/main/java/uk/gov/companieshouse/filinghistory/api/interceptor/RequestLoggingInterceptor.java:38]: See above

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+	rm -f ./$(artifact_name)-*.zip
+	rm -rf ./build-*
+	rm -rf ./build.log-*
+
+.PHONY: test
+test: test-integration test-unit
+
+.PHONY: test-unit
+test-unit: clean
+	mvn test
```
[the remark this answers is not in the document]
Comment (johncook-ch) [reply Makefile:17]: Done

**Review by cgregoryCH — APPROVED** [review]
Looks good

**Comment by sthompsonCH** [comment]
Did we not agree the plan for integration tests was a) to keep them in the `test` directory, and b) use the `IT.java` suffix rather than `ITest.java`? 

I'd be happy to go with an `itest` package like you have already done as that is not difficult to configure in the pom. However, I think we should be going with `IT.java` as the suffix for integration tests as it is easier to read and requires no extra configuration because it is the default config for the failsafe plugin.

What do you think?

**Review by sthompsonCH — APPROVED** [review]

**Pull request merged**